### PR TITLE
Propagate whether or not the trace was sampled for storage in a trace backend

### PR DESCRIPTION
**Pull request by TYsewyn**
Per the docs of https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry:

> The sampling decision of the trace associated with the log entry. True means that the trace resource name in the trace field was sampled for storage in a trace backend. False means that the trace was not sampled for storage when this log entry was written, or the sampling decision was unknown at the time.

Using the new reserved key `logging.googleapis.com/traceSampled` we can propagate whether or not the trace was sampled for storage in a trace backend.

### CODE REVIEW

**Comment by qingling128** [comment]
@davidbtucker FYI, this might overlap with some work you were planning to pick up.

File: test/plugin/test_out_google_cloud.rb [hunk 1]
```diff
@@ -330,6 +330,21 @@ def test_non_integer_timestamp
...
   end
 
+  # Will only work as "expected" with REST/JSON, but not with gRPC.
+  # The output of the gRPC result will be converted to JSON
+  # to get the assertions to work in the base test class.
+  # The problem with this is that default values are being omitted to save space.
+  # Reading this json will construct a struct which doesn't contain the field 'traceSampled',
+  # thus returning nil when trying to access that field instead of the expected false.
+  # And since 'nil' != 'false' this test will fail for gRPC.
+  # See https://developers.google.com/protocol-buffers/docs/proto3#json
```
[the remark this answers is not in the document]
Comment (TYsewyn) [reply test/plugin/test_out_google_cloud.rb]: Oh I see. I only looked at the generated file from the protocol buffer and so I overlooked the actual implementation 😅

**Comment by jkohen** [comment]
@qingling128 what does it take to resolve this PR?

**Comment by qingling128** [comment]
This is done already by https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/pull/297.

**Pull request closed without merging**